packages/hooks/src/createRunHook.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Returns a function which, when invoked, will execute all callbacks
+ * registered to a hook of the specified type, optionally returning the final
+ * value of the call chain.
+ *
+ * @param  {Object}   hooks          Stored hooks, keyed by hook name.
+ * @param  {?bool}    returnFirstArg Whether each hook callback is expected to
+ *                                   return its first argument.
+ *
+ * @return {Function}                Function that runs hook callbacks.
+ */
+function createRunHook( hooks, returnFirstArg ) {
+	/**
+	 * Runs all callbacks for the specified hook.
+	 *
+	 * @param  {string} hookName The name of the hook to run.
+	 * @param  {...*}   args     Arguments to pass to the hook callbacks.
+	 *
+	 * @return {*}               Return value of runner, if applicable.
+	 */
+	return function runHooks( hookName, ...args ) {
+		const handlers = hooks[ hookName ];
+		let maybeReturnValue = args[ 0 ];
+
+		if ( ! handlers ) {
```

Comment:
Currently this will trigger when a hook run is attempted but no callbacks have ever been added for that hook.

However, `runs` should increment if no handlers have ever been set, or if they have all been cleared, good catch there.  This will match PHP behavior.